src/name_bazaar/server/db_sync.cljs

```diff
@@ -131,6 +158,21 @@
                                                   {:from-block 0 :to-block "latest"}
                                                   (partial on-offering-bid server-state))]))
 
-(defn stop-syncing! []
+(defn stop-node-syncing! []
   (stop-watching-filters!)
   (reset! event-filters []))
+
+(defn start-syncing! [server-state]
+  (start-node-watchdog! @server-state
+                        (fn []
+                          (d0x-effects/create-db! server-state)
+                          (start-node-syncing! @server-state)
+                          (email-listeners/setup-event-listeners! server-state))
+                        (fn []
+                          (stop-node-syncing!)
+                          (email-listeners/stop-event-listeners! server-state))))
```

Comment:
It's probably best separation to tangle db_sync and email-listeners together. db_sync should take care only of in-memory db. Shouldn't know about email-listeners at all. I suggest you create ns `name-bazaar.server.watchdog` where you put all this watchdog logic and you'll require both db_sync and email-listeners there